<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      empty_world_patrolling.launch
\authors   Antonio Matta <antonio.matta@upm.es>
\copyright Copyright (c) 2017, Universida Politecnica de Madrid., All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that
the following conditions are met:
 * Redistributions of source code must retain the above copyright notice, this list of conditions and the
   following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
   following disclaimer in the documentation and/or other materials provided with the distribution.
 * Neither the name of Clearpath Robotics nor the names of its contributors may be used to endorse or promote
   products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

Launch file to send control waypoints to the fleet of robots.

Version: 1.0 $Id$: $file$ 2017-05-31
-->
<launch>
	<arg name="robot1" default="true"/>
	<arg name="robot2" default="false"/>
	<arg name="robot3" default="false"/>
  
  <!-- ROBOT 1 -->
  <group ns="summit1" if="$(arg robot1)">
		<node pkg="mrs_navigation" name="patrol_fsm" type="patrol_fsm.py" respawn="false"  output="screen">
			<param name="robot_name" value="/summit1" />
			<param name="waypoints" value="[[-8.0, 8.5, 90.0], [9.0, 8.5, 180.0], [-8.0, 8.5, -90.0], [-9.0, -9.0, 90.0]]"/>
			<!--<param name="waypoints" value="[[-8.0, 9.0, 90.0], [9.0, 9.0, -90.0], [9.0, -9.0, -90.0], [-9.0, -9.0, 90.0]]"/> -->
    </node>
  </group>
  

  <!-- ROBOT 2 -->
  <group ns="summit2" if="$(arg robot2)">
		<node pkg="mrs_navigation" name="patrol_fsm" type="patrol_fsm.py" respawn="false"  output="screen">
			<param name="robot_name" value="/summit2" />
			<param name="waypoints" value="[[9.0, -9.0, -90.0], [-9.0, -9.0, 0.0], [9.0, -9.0, 90.0], [9.0, 9.0, -90.0]]"/>
			<!--<param name="waypoints" value="[[9.0, -9.0, -90.0], [-9.0, -9.0, 90.0], [-8.0, 9.0, 90.0], [9.0, 9.0, -90.0]]"/> -->
    </node>
  </group> 
   
   <!-- ROBOT 3 -->
  <group ns="summit3" if="$(arg robot3)">
		<node pkg="mrs_navigation" name="patrol_fsm" type="patrol_fsm.py" respawn="false"  output="screen">
			<param name="robot_name" value="/summit3" />
			<param name="waypoints" value="[[7.0, -3.0, 90.0], [7.0, 3.0, 180.0], [-7.0, 3.0, -90.0], [-7.0, -3.0, 0.0]]" />
    </node>
  </group> 

</launch>
